packages/angular-cli/blueprints/module/index.js

```diff
@@ -58,7 +58,9 @@ module.exports = {
   },
 
   afterInstall: function (options) {
-    options.entity.name = path.join(this.dasherizedModuleName, this.dasherizedModuleName);
+    var componentPath = path.join(this.generatePath, this.dasherizedModuleName);
```

Comment:
If you're only going to have `this.dasherizedModuleName` once, then you need to turn `options.flat` to false. The default for components is that they're in their own directory. I'd suggest using `options.flat = false`. And once you make that change you will also need to update your tests.